cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisioningService.java

```diff
@@ -517,6 +541,38 @@ private SSHContext createSSHContext(@Nullable SecureKeyInfo keyInfo) {
     return keyInfo == null ? null : new DefaultSSHContext(locationFactory, keyInfo);
   }
 
+  private ProvisionerContext createContext(ProgramRunId programRunId, String userId, Map<String, String> properties,
+                                           @Nullable SSHContext sshContext) {
+    Map<String, String> evaluated = evaluateMacros(secureStore, userId, programRunId.getNamespace(), properties);
+    return new DefaultProvisionerContext(programRunId, evaluated, sparkCompat, sshContext);
+  }
+
+  @VisibleForTesting
```

Comment:
would be good to add javadoc here, mainly around the fact that the macro substitution using secure store will only apply for properties values and how userId is used, since this is the main logic of this change.